### ISSUE TITLE
panic: interface conversion: interface {} is nil, not map[string]interface {} in aws_config_config_rule

### ISSUE THREAD

**Issue by joseph-wortmann** [terraform-providers/terraform-provider-aws#5848]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.8
+ provider.aws v1.35.0
+ provider.null v1.0.0

### Affected Resource(s)

* aws_config_config_rule

### Terraform Configuration Files

This is a simplified call to the resource which illustrates the problem. Basically, when the `scope` is presented with empty strings for the attributes, the provider panics and crashes Terraform. The use case for doing this is the embedding of the `aws_config_rule_resource` in a module (which I am doing), and allowing the module users to optionally specify a `scope`. Works great when the scope is specified, obviously not so great when it is not.

```hcl
resource "aws_config_config_rule" "rule" {
  count             = "${var.rules_count}"
  depends_on        = [
    "aws_config_configuration_recorder.config"
  ]
  input_parameters  = "${lookup(var.input_parameters, element(var.rules, count.index), "")}"
  name              = "${element(var.rules, count.index)}"
  scope {
    tag_key = ""
    tag_value = ""
  }
  source {
    owner             = "AWS"
    source_identifier = "${lookup(local.source_identifiers, element(var.rules, count.index))}"
  }
}
```

### Panic Output

[crash.log](https://github.com/terraform-providers/terraform-provider-aws/files/2375373/crash.log)

### Expected Behavior

I believe that the `aws_config_config_rule` should treat empty strings in the `scope` attributes as if those attributes were never referenced.

### Actual Behavior

panic

### Steps to Reproduce

Simply try to create an `aws_config_config_rule` resources that references the `tag_key` and `tag_value` in `scope`, and supply those with empty strings. Then wait for the panic.

### References

I couldn't find any.

**Comment by bflad** [comment]
Thanks for the report @joseph-wortmann and sorry you ran into trouble. Submitted a pull request to fix this crash that verifies the empty string configuration should work as expected afterwards: #5852

**Comment by joseph-wortmann** [comment]
No need to apologize. Software has bugs. I'm a software guy too. Thank you for jumping on this so quickly!!!

**Comment by bflad** [comment]
The fix for this has been merged into master and will release with version 1.37.0 of the AWS provider, likely middle of next week. 👍

**Comment by bflad** [comment]
This has been released in [version 1.37.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1370-september-19-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
